spring-security/src/main/java/com/sap/cloud/security/spring/token/SpringSecurityContext.java

```diff
@@ -0,0 +1,107 @@
+package com.sap.cloud.security.spring.token;
+
+import com.sap.cloud.security.spring.token.authentication.HybridJwtDecoder;
+import com.sap.cloud.security.spring.token.authentication.XsuaaTokenAuthorizationConverter;
+import com.sap.cloud.security.token.AccessToken;
+import com.sap.cloud.security.token.Token;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.util.Assert;
+
+import javax.annotation.Nullable;
+
+/**
+ * This is an alternative way of accessing jwt tokens of type {@link Token} or
+ * {@link AccessToken} in context of an application using
+ * spring-security-oauth2.
+ *
+ * It uses the {@link SecurityContextHolder} to access Spring's
+ * {@link SecurityContext} and can therefore used also in context of
+ * asynchronous threads.
+ */
+public class SpringSecurityContext {
```

Comment:
Will a representation of the token also be available via the spring-independent [SecurityContext](https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/java-api/src/main/java/com/sap/cloud/security/token/SecurityContext.java) when using this library? This would be great, as it would help libraries to retrieve the token with a single uniform API, independent if the application is using Spring or not.

Reply:
not yet but good point - maybe this can be solved with a custom https://docs.spring.io/spring-security/site/docs/4.0.x/apidocs/org/springframework/security/core/context/SecurityContextHolderStrategy.html

Reply:
see here:
https://github.com/SAP/cloud-security-xsuaa-integration/pull/450/commits/6b601ebbbd7925af4523842574fa2adbff57866e

and here: https://jtrack.wdf.sap.corp/browse/SECAUTH-1270

Reply:
Looks good! I think auto-configuration for this would be very helpful :)

Reply:
Done with that: https://github.com/SAP/cloud-security-xsuaa-integration/pull/450/commits/14a22ab7ee880555a252a13d594cd2cafe12be43